src/test/app/Escrow_test.cpp

```diff
@@ -1277,6 +1277,134 @@ struct Escrow_test : public beast::unit_test::suite
         }
     }
 
+    void testDeletedDestination()
+    {
+        // Make sure an Escrow behaves as expected if its destination is
+        // deleted from the ledger.  This can only happen if fix1523 is 
+        // not enabled.  fix1523 is what adds the Escrow to the Escrow's
+        // destination directory.
+        testcase ("Deleted destination");
+
+        using namespace jtx;
+        using namespace std::chrono;
+
+        Account const alice {"alice"};
+        Account const bruce {"bruce"};
+        Account const carol {"carol"};
+        Account const daria {"daria"};
+        Account const evita {"evita"};
+        Account const fiona {"fiona"};
+        Account const grace {"grace"};
+
+        Env env(*this, supported_amendments() - fix1523);
+
+        env.fund(XRP(5000), alice, bruce, carol, daria, evita, fiona, grace);
+        env.close();
+
+        // Because fix1523 is not in play, bruce does not have a back link
+        // from his directory to the escrow.  So we should be able to delete
+        // bruce's account even though he is the destination of an escrow.
+        std::uint32_t const aliceEscrowSeq {env.seq(alice)};
+        env(escrow(alice, bruce, XRP(1000)), finish_time(env.now() + 1s));
+
+        // Similar to bruce, we should be able to delete daria's account.
+        std::uint32_t const carolEscrowSeq {env.seq(carol)};
+        env(escrow(carol, daria, XRP(1000)),
+            finish_time(env.now() + 1s), cancel_time(env.now() + 2s));
+        env.close();
+
+        // Now engage fix1523 so any additional escrows we make will have
+        // back links from both the source and the destination.
+        env.enableFeature (fix1523);
+        env.close();
+
+        // Neither evita not fiona should be able to delete their accounts
+        // once this escrow is created.
+        std::uint32_t const evitaEscrowSeq {env.seq(evita)};
+        env(escrow(evita, fiona, XRP(1000)), finish_time(env.now() + 1s));
+        env.close();
+
+        // Close enough ledgers to be able to delete any of the accounts.
+        std::uint32_t const ledgerCount {
+            env.current()->seq() + 257 - env.seq (alice)};
```

Comment:
I don't believe this is the correct ledgerCount. See this comment: https://github.com/ripple/rippled/pull/3093#discussion_r331691858 and this patch: https://github.com/seelabs/rippled/commit/f310e26b5e73ce5f0097521a574e7e7bd7ec9e07

(note that patch fixes other instances of this bug, but does not fix it here)